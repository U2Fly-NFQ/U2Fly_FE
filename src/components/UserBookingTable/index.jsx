import React, { useState } from 'react'
import { UserBookingDetail } from '../index'
import { Button, Space, Table } from 'antd'
import { bookingStatus } from '../../Constants'
import { useNavigate } from 'react-router-dom'

function UserBookingTable({ loading, data }) {
  //initiation
  const [expandedRowKeys, setExpandedRowKeys] = useState([])
  const navigate = useNavigate()
  //Data for UI
  const bookingListColumn = [
    {
      title: 'Booking ID',
      dataIndex: 'id',
      align: 'center',
      sorter: (a, b) => a.id - b.id,
    },
    {
      title: 'Journey',
      dataIndex: 'flights',
      align: 'center',
      render: (_, { flights }) => (
        <>
          {flights[0].departure}
          {flights.length === 2 ? (
            <i
              style={{ padding: '0 10px' }}
              className="fa-solid fa-arrow-right-arrow-left"
            ></i>
          ) : (
            <i
              style={{ padding: '0 10px' }}
              className="fa-solid fa-arrow-right-long"
            ></i>
          )}
          {flights[0].arrival}
        </>
      ),
    },
    {
      title: 'Booking Amount',
      dataIndex: 'totalPrice',
      align: 'center',
      sorter: (a, b) => a.totalPrice - b.totalPrice,
    },
    {
      title: 'Status',
      dataIndex: 'status',
      align: 'center',
      sorter: (a, b) => a.status.length - b.status.length,
    },
    {
      title: 'Action',
      key: 'action',
      width: '100px',
      align: 'center',
      render: (_, record) => (
        <Space>
          {/* eslint-disable-next-line react/jsx-no-undef */}
          {record.status === bookingStatus['0'] && (
            <Button
              type="default"
              shape="default"
              // onClick={() => ()}
            >
              Cancel
            </Button>
          )}
          {(record.status === bookingStatus['1'] ||
            record.status === bookingStatus['2']) && (
            <Button
              type="primary"
              shape="default"
              onClick={() => navigate('/')}
            >
              Booking Again
            </Button>
          )}
        </Space>
      ),
    },
  ]
  const onExpandRowKey = (expanded, record) => {
    expanded ? setExpandedRowKeys([record.id]) : setExpandedRowKeys([])
  }

  return (
    <Table
      columns={bookingListColumn}
      rowKey={(record) => record.id}
      expandable={{
        expandedRowKeys: expandedRowKeys,
        onExpand: onExpandRowKey,
        expandedRowRender: (record) => (
          <UserBookingDetail detailData={record} />
        ),
      }}
      dataSource={data}
      loading={loading}
      scroll={{
        x: 'max-content',
      }}
    />
  )
}

export default UserBookingTable
