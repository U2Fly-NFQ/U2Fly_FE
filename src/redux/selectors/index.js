export * from './auth'
export * from './bookingFlightSelector'
export * from './ticket'
