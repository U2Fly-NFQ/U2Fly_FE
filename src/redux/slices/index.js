export * from './searchSlice'
export * from './filterSlice'
export * from './flightSlice'
