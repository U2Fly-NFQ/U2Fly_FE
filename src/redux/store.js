import { configureStore } from '@reduxjs/toolkit'
import bookingFlightsSlice from './slices/bookingFlightsSlice'

// import loginSlice from './slices/loginSlice'
import flightSlice from './slices/flightSlice'
import airportSlice from './slices/airportSlice'

const store = configureStore({
  reducer: {
    // login: loginSlice.reducer,
    bookingFlight: bookingFlightsSlice.reducer,
    flights: flightSlice.reducer,
    airports: airportSlice.reducer,
  },
})

export default store
