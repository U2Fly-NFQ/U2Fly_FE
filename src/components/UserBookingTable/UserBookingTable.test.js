import renderer from 'react-test-renderer'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import UserBookingTable from './'
import { store } from '../../redux/store'
import { Provider } from 'react-redux'
describe('Navigation bar test', () => {
  it('renders correctly', () => {
    let tickets = [
      {
        id: 21,
        passenger: {
          name: 'Sang Nguyen',
          gender: 'Male',
          birthday: {
            date: '2000-01-24 00:00:00.000000',
            timezone_type: 3,
            timezone: 'UTC',
          },
          address: 'Can Tho',
          identification: '222',
          accountId: 2,
          email: 'sang@gg.com',
        },
        totalPrice: 98.01,
        ticketOwner: 'Sang Nguyen',
        paymentId: 'pi_3LJk17GLKOTVyCxb1DVaqeTo',
        email: 'sang@gg.com',
        discount: 0,
        seatType: 'Economy',
        createdAt: '2022-07-09 20:12:51',
        status: 'Booking successful',
        flights: [
          {
            id: 8,
            code: 'VN757',
            arrival: {
              id: 5,
              iata: 'PQC',
              name: 'Phu Quoc International Airport',
              city: 'Kien Giang',
              image: 'google.com',
            },
            departure: {
              id: 2,
              iata: 'HAN',
              name: 'Noi Bai International Airport',
              city: 'Ha Noi',
              image: 'google.com',
            },
            startTime: '20:00:00',
            startDate: '2022-07-10',
            duration: 1.5,
            airplane: {
              id: 4,
              name: 'Boeing 787-9 Dreamliner',
            },
            airline: {
              id: 2,
              name: 'Bamboo Airways',
              icao: 'BAV',
              image:
                'https://cdn.haitrieu.com/wp-content/uploads/2022/01/Logo-Bamboo-Airways-H.png',
            },
            seat: [
              {
                id: 1,
                name: 'Economy',
                price: 99,
                seatAvailable: 66,
                discount: 0.1,
              },
              {
                id: 2,
                name: 'Business',
                price: 159,
                seatAvailable: 15,
                discount: 0.1,
              },
            ],
            ticketFlight: {
              id: 16,
              isRating: null,
              createdAt: {
                date: '2022-07-09 20:12:51.000000',
                timezone_type: 3,
                timezone: 'UTC',
              },
              flight: {
                __initializer__: null,
                __cloner__: null,
                __isInitialized__: true,
              },
              ticket: {},
            },
          },
        ],
      },
    ]
    let ticketProcessedData = []

    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <UserBookingTable
              data={ticketProcessedData}
              loading={true}
              //   onCancel={handleCancelBooking}
            />
          </Provider>
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
