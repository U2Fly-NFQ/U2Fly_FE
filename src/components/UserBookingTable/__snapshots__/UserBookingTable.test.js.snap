// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar test renders correctly 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div>
      <div
        aria-busy={true}
        aria-live="polite"
        className="ant-spin ant-spin-spinning"
      >
        <span
          className="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
    <div
      className="ant-spin-container ant-spin-blur"
    >
      <div
        className="ant-table ant-table-empty ant-table-scroll-horizontal"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={
              Object {
                "overflowX": "auto",
                "overflowY": "hidden",
              }
            }
          >
            <table
              style={
                Object {
                  "minWidth": "100%",
                  "tableLayout": "auto",
                  "width": "max-content",
                }
              }
            >
              <colgroup>
                <col
                  className="ant-table-expand-icon-col"
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "100px",
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell ant-table-row-expand-icon-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={Object {}}
                  />
                  <th
                    className="ant-table-cell ant-table-column-has-sorters"
                    colSpan={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                    tabIndex={0}
                  >
                    <div
                      className="ant-table-column-sorters"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <span
                        className="ant-table-column-title"
                      >
                        Booking ID
                      </span>
                      <span
                        className="ant-table-column-sorter ant-table-column-sorter-full"
                      >
                        <span
                          className="ant-table-column-sorter-inner"
                        >
                          <span
                            aria-label="caret-up"
                            className="anticon anticon-caret-up ant-table-column-sorter-up"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="caret-up"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                              />
                            </svg>
                          </span>
                          <span
                            aria-label="caret-down"
                            className="anticon anticon-caret-down ant-table-column-sorter-down"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="caret-down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Journey
                  </th>
                  <th
                    className="ant-table-cell ant-table-column-has-sorters"
                    colSpan={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                    tabIndex={0}
                  >
                    <div
                      className="ant-table-column-sorters"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <span
                        className="ant-table-column-title"
                      >
                        Booking Amount
                      </span>
                      <span
                        className="ant-table-column-sorter ant-table-column-sorter-full"
                      >
                        <span
                          className="ant-table-column-sorter-inner"
                        >
                          <span
                            aria-label="caret-up"
                            className="anticon anticon-caret-up ant-table-column-sorter-up"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="caret-up"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                              />
                            </svg>
                          </span>
                          <span
                            aria-label="caret-down"
                            className="anticon anticon-caret-down ant-table-column-sorter-down"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="caret-down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Status
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Action
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  aria-hidden="true"
                  className="ant-table-measure-row"
                  style={
                    Object {
                      "fontSize": 0,
                      "height": 0,
                    }
                  }
                >
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                </tr>
                <tr
                  className="ant-table-placeholder"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                >
                  <td
                    className="ant-table-cell"
                    colSpan={6}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <div
                      className="ant-table-expanded-row-fixed"
                      style={
                        Object {
                          "left": 0,
                          "overflow": "hidden",
                          "position": "sticky",
                          "width": 0,
                        }
                      }
                    >
                      <div
                        className="ant-empty ant-empty-normal"
                      >
                        <div
                          className="ant-empty-image"
                        >
                          <svg
                            className="ant-empty-img-simple"
                            height="41"
                            viewBox="0 0 64 41"
                            width="64"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fillRule="evenodd"
                              transform="translate(0 1)"
                            >
                              <ellipse
                                className="ant-empty-img-simple-ellipse"
                                cx="32"
                                cy="33"
                                rx="32"
                                ry="7"
                              />
                              <g
                                className="ant-empty-img-simple-g"
                                fillRule="nonzero"
                              >
                                <path
                                  d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                />
                                <path
                                  className="ant-empty-img-simple-path"
                                  d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                        <div
                          className="ant-empty-description"
                        >
                          No Data
                        </div>
                      </div>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
